bls/src/keypair.rs

```diff
@@ -75,6 +88,26 @@ impl SecretKey {
     }
 }
 
+impl TryFrom<&[u8]> for SecretKey {
+    type Error = BlsError;
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != BLS_SECRET_KEY_SIZE {
+            return Err(BlsError::ParseFromBytes);
+        }
+        // unwrap safe due to the length check above
+        let scalar: Option<Scalar> = Scalar::from_bytes_le(bytes.try_into().unwrap()).into();
+        scalar.ok_or(BlsError::FieldDecode).map(Self)
+    }
+}
+
+impl From<&SecretKey> for [u8; BLS_SECRET_KEY_SIZE] {
+    fn from(secret_key: &SecretKey) -> Self {
+        let mut bytes = [0u8; BLS_SECRET_KEY_SIZE];
+        bytes.copy_from_slice(secret_key.0.to_bytes_le().as_slice());
+        bytes
```

Comment:
This implementation is a bit redundant -- `to_bytes_le()` already returns a properly-sized array, so you can just do
```suggestion
        secret_key.0.to_bytes_le()
```

Reply:
Oh oops. Thanks for the catch!